File: services/laravel-svc/overlay/app/Http/Controllers/HealthController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Support\Facades\DB;

class HealthController extends Controller
{
    // Liveness: cheap, no DB, the process is up.
    public function live()
    {
        return response()->json(['status' => 'UP']);
    }

    // Readiness: gated on a live DB round-trip.
    public function ready()
    {
        try {
            DB::select('SELECT 1');
            return response()->json(['status' => 'UP']);
        } catch (\Throwable $e) {
            return response()->json(['status' => 'DOWN'], 503);
        }
    }
}

File: services/rails-svc/app/controllers/health_controller.rb
class HealthController < ApplicationController
  # Liveness: cheap, no DB, the process is up.
  def live
    render json: { status: "UP" }
  end

  # Readiness: gated on a live DB round-trip.
  def ready
    ActiveRecord::Base.connection.execute("SELECT 1")
    render json: { status: "UP" }
  rescue StandardError
    render json: { status: "DOWN" }, status: :service_unavailable
  end
end
